fastestimator/estimator.py

```diff
@@ -384,10 +400,10 @@ def _configure_loader(self, loader: Union[DataLoader, tf.data.Dataset]) -> Union
             new_loader = tf.data.Dataset.from_generator(lambda: loader, data_type, output_shapes=data_shape)
             new_loader = new_loader.prefetch(1)
         if isinstance(new_loader, tf.data.Dataset):
-            if self.system.max_train_steps_per_epoch and self.system.mode == "train":
-                new_loader = new_loader.take(self.system.max_train_steps_per_epoch)
-            if self.system.max_eval_steps_per_epoch and self.system.mode == "eval":
-                new_loader = new_loader.take(self.system.max_eval_steps_per_epoch)
+            if self.system.train_steps_per_epoch and self.system.mode == "train":
+                new_loader = new_loader.take(self.system.train_steps_per_epoch)
```

Comment:
what will happen here if the user tries to expand a tf.dataset that they are providing (instead of using one of our FE datasets)?